src/main/java/duke/command/DeletePatientCommand.java

```diff
@@ -49,33 +51,44 @@ public DeletePatientCommand(String deletedPatientInfo) throws DukeException {
      * @throws DukeException .
      */
     @Override
-    public void execute(PatientTaskList patientTask, TaskManager tasks, PatientManager patientManager,
+    public void execute(PatientTaskList patientTaskList, TaskManager taskManager, PatientManager patientManager,
                         Ui ui, PatientTaskStorage patientTaskStorage, TaskStorage taskStorage,
                         PatientStorage patientStorage) throws DukeException {
         ;
+        boolean toDelete = false;
+        ArrayList<PatientTask> patientTask = new ArrayList<>();
+        Patient patientToBeDeleted = null;
         if (id != 0) {
-            Patient patientToBeDeleted = patientManager.getPatient(id);
-            boolean toDelete = ui.confirmPatientToBeDeleted(patientToBeDeleted);
-            if (toDelete) {
-                patientManager.deletePatient(id);
-                ui.patientDeleted();
-                patientStorage.save(patientManager.getPatientList());
-            }
+            patientToBeDeleted = patientManager.getPatient(id);
+            patientTask = patientTaskList.getPatientTask(id);
+            ui.showPatientInfo(patientToBeDeleted);
         } else {
             ArrayList<Patient> patientsWithSameName = patientManager.getPatientByName(deletedPatientInfo);
             ui.patientsFoundByName(patientsWithSameName, deletedPatientInfo);
             if (patientsWithSameName.size() >= 1) {
                 int numberChosen = ui.choosePatientToDelete(patientsWithSameName.size());
                 if (numberChosen >= 1) {
-                    boolean toDelete = ui.confirmPatientToBeDeleted(patientsWithSameName.get(numberChosen - 1));
-                    if (toDelete) {
-                        patientManager.deletePatient(patientsWithSameName.get(numberChosen - 1).getID());
-                        ui.patientDeleted();
-                        patientStorage.save(patientManager.getPatientList());
-                    }
+                    patientToBeDeleted = patientsWithSameName.get(numberChosen - 1);
+                    patientTask = patientTaskList.getPatientTask(patientToBeDeleted.getID());
+                    ui.showPatientInfo(patientToBeDeleted);
                 }
             }
         }
+        if (patientTask.size() > 0) {
+            ArrayList<Task> tempTask = new ArrayList<>();
+            for (PatientTask temppatientTask : patientTask) {
+                tempTask.add(taskManager.getTask(temppatientTask.getTaskID()));
+            }
+            ui.patientTaskFound(patientToBeDeleted, patientTask, tempTask);
+            toDelete = ui.confirmPatientToBeDeleted(patientToBeDeleted, true);
+        } else {
+            toDelete = ui.confirmPatientToBeDeleted(patientToBeDeleted,false);
+        }
+        if (toDelete) {
+            patientManager.deletePatient(patientToBeDeleted.getID());
+            ui.patientDeleted();
```

Comment:
The patientManager deletes the Patient, but where/how does the taskList delete the patient's tasks?